File: demo-fraud-webapp/src/app/interfaces/Alert.ts
import {Transaction} from "./Transaction"; // 导入 Transaction 接口，表示触发警报的交易数据结构。
import {RefObject} from "react"; // 从 React 库导入 RefObject 类型，用于引用 React 元素。
import {RulePayload} from "./Rule"; // 导入 RulePayload 接口，表示规则的详细数据结构。

/**
 * 警报接口定义，用于描述触发的警报及其相关信息。
 */
export interface Alert {
    alertId: string; // 警报的唯一标识符，用于区分不同的警报。
    ruleId: number; // 触发警报的规则的唯一标识符。
    violatedRule: RulePayload; // 被违反的规则的详细信息，使用 RulePayload 接口描述规则内容。
    triggeringValue: number; // 触发警报的值，表示实际的交易或计算值超过了规则限制。
    triggeringEvent: Transaction; // 触发警报的交易事件，包含交易的详细信息。
    ref: RefObject<HTMLDivElement>; // React 引用对象，指向与警报相关的 HTMLDivElement，便于操作相关 DOM 元素。
}

File: fraud-webapp/src/main/java/com/ververica/demo/backend/model/Alert.java
package com.ververica.demo.backend.model;

import com.ververica.demo.backend.datasource.Transaction;

import java.math.BigDecimal;

import lombok.AllArgsConstructor;
import lombok.Data;

/**
 * 警报模型，用于表示触发了某个规则的警报详细信息。
 */
@Data // Lombok注解，自动生成所有属性的getter和setter方法，以及toString、equals和hashCode方法。
@AllArgsConstructor // Lombok注解，自动生成包含所有字段的构造函数。
public class Alert {
    private Integer ruleId; // 规则ID，标识触发警报的规则。
    private String rulePayload; // 规则载荷，包含规则的具体内容。

    Transaction triggeringEvent; // 触发此警报的交易事件。
    BigDecimal triggeringValue; // 触发此警报的数值，可能基于交易金额或其他计算得出。
}
